server/src/main/java/org/opensearch/indices/replication/SegmentReplicator.java

```diff
@@ -217,7 +218,7 @@ protected void pruneCheckpointsUpToLastSync(final IndexShard indexShard) {
             );
 
             if (existingCheckpointStats != null && !existingCheckpointStats.isEmpty()) {
-                existingCheckpointStats.keySet().removeIf(key -> key < segmentInfoVersion);
+                existingCheckpointStats.keySet().removeIf(key -> key <= segmentInfoVersion);
```

Comment:
Would this cause any issues will rolling upgrades from the BWC perspective?

Reply:
no, this doesn't change any transport level api, only computation of stats locally on the node.